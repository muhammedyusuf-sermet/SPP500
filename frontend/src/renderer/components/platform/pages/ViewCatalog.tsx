import * as React from 'react';
import * as PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';

import {Monster} from './view_catalog/Monster';

import '../../../css/platform/pages/view_catalog.css';
import { CharacterList } from './view_catalog/CharacterList';

type AppProps = {}

function TabContainer(props: TabContainer.propTypes) {
	return (
		<Typography component="div" style={{ padding: 8 * 3 }}>
			{props.children}
		</Typography>
	);
}

TabContainer.propTypes = {
	children: PropTypes.node.isRequired,
};

export const ViewCatalog: React.StatelessComponent<{}> = () => {
	const [value, setValue] = React.useState(0);

	function handleChange(event: AppProps, newValue: number) {
		setValue(newValue);
	}

	return (
		<div className="view-catalog-container">
			<h1 className="page-title">View Catalog Items</h1>
			<div className= "tab-root">
			<AppBar position="static">
				<Tabs value={value} onChange={handleChange}>
					<Tab label="Monsters" />
					<Tab label="Equipment" />
					<Tab label="Locations" />
					<Tab label="Buildings" />
					<Tab label="Characters" />
					<Tab label="Spells" />
				</Tabs>
			</AppBar>
			{value === 0 && <TabContainer><Monster/></TabContainer>}
			{value === 1 && <TabContainer>Equipment</TabContainer>}
			{value === 2 && <TabContainer>Locations</TabContainer>}
			{value === 3 && <TabContainer>Buildings</TabContainer>}
			{value === 4 && <TabContainer><CharacterList/></TabContainer>}
			{value === 5 && <TabContainer>Spells</TabContainer>}
			</div>
		</div>
	);
}
