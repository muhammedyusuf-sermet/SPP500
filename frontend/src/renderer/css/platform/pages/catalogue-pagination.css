#paginated-catalogue-navigation a {
  text-decoration: none;
  display: inline-block;
  padding: 8px 16px;
  margin-bottom: 20px;
}

#paginated-catalogue-navigation a:hover {
  background-color: #ddd;
  color: black;
}

#paginated-catalogue-navigation .previous {
  background-color: #f1f1f1;
  color: black;
}

#paginated-catalogue-navigation .next {
  background-color: #4CAF50;
  color: white;
}

#paginated-catalogue-navigation .round {
  border-radius: 50%;
}