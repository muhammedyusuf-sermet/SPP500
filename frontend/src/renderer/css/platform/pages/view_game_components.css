.view-game-components-container {
	text-align: center;
}
.page-title {
	padding: 5px 15px;
	display: inline-block;
}

.view-game-components-container .tab-root {
	flex-grow: 1;
}

.view-game-components-container .app-bar {
	position: relative;
}

.view-game-components-container .layout {
	width: auto;
	margin-left: 2%;
	margin-right: 2%;
}

.view-game-components-container .card-grid {
	padding: 8px 0;
}

.view-game-components-container .card {
	height: 100%;
	display: flex;
	flex-direction: column;
}

.view-game-components-container .card-media {
	padding-top: 56.25%; /* 16:9 */
}

.view-game-components-container .card-content {
	flex-grow: 1;
}
