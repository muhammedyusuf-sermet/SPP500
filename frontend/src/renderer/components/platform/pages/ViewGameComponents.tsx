import * as React from 'react';

import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

import {EncounterList} from './view_game_components/EncounterList';
import {Campaign} from './view_game_components/Campaign';

import '../../../css/platform/pages/view_game_components.css';
import { RouteComponentProps, Route } from 'react-router';
import { CharacterList } from './view_game_components/CharacterList';

export interface IViewGameComponentsProps extends RouteComponentProps{
}

export interface IViewGameComponentsState {
	value: string;
}

export class ViewGameComponents extends React.Component<IViewGameComponentsProps, IViewGameComponentsState> {
	constructor(props: IViewGameComponentsProps) {
		super(props);
		this.state = {
			value: props.location.pathname.split('/').pop() || ''
		}
	}

	componentWillReceiveProps(nextProps: IViewGameComponentsProps) {
		this.setState({
			value: nextProps.location.pathname.split('/').pop() || ''
		});
	}

	render() {
		return (
			<div className="view-game-components-container">
				<h1 className="page-title">View Catalog Items</h1>
				<div className= "tab-root">
					<AppBar position="static">
						<Route render={({history}) => (
							<Tabs value={this.state.value} onChange={(event, newValue) => {
									history.replace('/game_components/'+newValue);
									this.setState({value: newValue});
								}}>
								<Tab label="Encounters" value='encounters' />
								<Tab label="Campaigns" value='campaigns' />
								<Tab label="Characters" value='characters' />
							</Tabs>
						)} />
					</AppBar>
					<Route path={`${this.props.match.url}/encounters`} component={EncounterList} />
					<Route path={`${this.props.match.url}/campaigns`} component={Campaign} />
					<Route path={`${this.props.match.url}/characters`} component={CharacterList} />
				</div>
			</div>
		);
	}
}
