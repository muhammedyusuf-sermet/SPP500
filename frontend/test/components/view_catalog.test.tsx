import * as React from 'react';
import {shallow} from 'enzyme';
import {ViewCatalog} from '../../src/renderer/components/platform/pages/ViewCatalog';
import { Route } from 'react-router';

jest.mock('../../src/cookie');

describe('Test the View Catalog Page', () => {
	const viewCatalogInstance = shallow(<Route path="/catalog" component={ViewCatalog} />);

	it('renders without crashing', () => {
		expect(viewCatalogInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(viewCatalogInstance).toMatchSnapshot();
	});

	it('renders a login-header-container classed component', () => {
		expect(viewCatalogInstance.find('.view-catalog-container')).toExist();
	});

});
