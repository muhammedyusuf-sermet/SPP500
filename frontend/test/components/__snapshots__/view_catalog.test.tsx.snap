// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the View Catalog Page renders correctly when the page is loaded 1`] = `
<div
  className="view-catalog-container"
>
  <h1
    className="page-title"
  >
    View Catalog Items
  </h1>
  <div
    className="tab-root"
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Tabs)
        onChange={[Function]}
        value={0}
      >
        <WithStyles(Tab)
          label="Monsters"
        />
        <WithStyles(Tab)
          label="Equipment"
        />
        <WithStyles(Tab)
          label="Locations"
        />
        <WithStyles(Tab)
          label="Buildings"
        />
        <WithStyles(Tab)
          label="Characters"
        />
        <WithStyles(Tab)
          label="Spells"
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
    <TabContainer>
      <Monster />
    </TabContainer>
  </div>
</div>
`;
